lib/components/database_config_form/database_config_form.dart

```diff
@@ -0,0 +1,253 @@
+import 'package:expense_manager/components/database_config_form/constants.dart';
+import 'package:expense_manager/components/form_helpers/form_dropdown.dart';
+import 'package:expense_manager/components/form_helpers/form_field.dart';
+import 'package:expense_manager/utils/database_config_store/database_config_store.dart';
+import 'package:expense_manager/utils/database_manager/database_manager.dart';
+import 'package:expense_manager/utils/logger/logger.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class DatabaseConfigForm extends StatefulWidget {
+  const DatabaseConfigForm({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _DatabaseConfigFormState();
+}
+
+class _DatabaseConfigFormState extends State<DatabaseConfigForm> {
+  // Create a global key that uniquely identifies the Form widget
+  // and allows validation of the form.
+  //
+  // Note: This is a GlobalKey<FormState>,
+  // not a GlobalKey<MyCustomFormState>.
+  final _formKey = GlobalKey<FormState>();
+  var isSubmitting = false;
+  var isLoadingConfig = true;
+  var dbConfigStore = DatabaseConfigStore();
+  Map<String, TextEditingController> formControllerMap = {
+    hostTextFormFieldLabel: TextEditingController(),
+    portTextFormFieldLabel: TextEditingController(),
+    nameTextFormFieldLabel: TextEditingController(),
+    usernameTextFormFieldLabel: TextEditingController(),
+    passwordTextFormFieldLabel: TextEditingController(),
+    sslModeDropdownFieldLabel: TextEditingController(),
+  };
+  var configToFieldMap = {
+    DatabaseConfigStore.dbHostKey: hostTextFormFieldLabel,
+    DatabaseConfigStore.dbPortKey: portTextFormFieldLabel,
+    DatabaseConfigStore.dbNameKey: nameTextFormFieldLabel,
+    DatabaseConfigStore.dbUsernameKey: usernameTextFormFieldLabel,
+    DatabaseConfigStore.dbPasswordKey: passwordTextFormFieldLabel,
+    DatabaseConfigStore.sslModeKey: sslModeDropdownFieldLabel,
+  };
+
+  @override
+  void initState() {
+    super.initState();
+    loadConfig();
+  }
+
+  Future<void> loadConfig() async {
+    for (var e in configToFieldMap.entries) {
+      await dbConfigStore.getConfigValue(e.key).then((val) {
+        if (val != null) {
+          formControllerMap[e.value]?.text = val;
+        }
+      });
+    }
+    setState(() {
+      isLoadingConfig = false;
+    });
+  }
+
+  String? checkEmptyInput(String? value) {
+    if (value == null || value.isEmpty) {
+      return 'Please enter a value';
+    }
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    return isLoadingConfig
+        ? const Center(child: CircularProgressIndicator())
+        : Form(
+            key: _formKey,
+            child: Column(
```

Comment:
This form is pretty long. Change the keyboard to overlay instead of collapse the fields together. May need to wrap the column in a scrollable.

<img width="342" alt="Screenshot 2024-09-22 at 8 00 44 PM" src="https://github.com/user-attachments/assets/42c8b1ad-ec8f-4b57-83fe-a349c5fb133a">

Reply:
Thanks for the suggestion! I plan on addressing all of these UI related issues in a near future PR 🙂. Tracking this at #15.

Reply:
LGTM then! 😎